spec/services/landing_page_finder_spec.rb

```diff
@@ -0,0 +1,126 @@
+# frozen_string_literal: true
+
+describe LandingPageFinder do
+  describe '#landing_page' do
+    include Rails.application.routes.url_helpers
+
+    let(:user) { nil }
+    let(:return_to) { nil }
+    let(:finder) { described_class.new(user, return_to) }
+
+    context 'with a user who is not approved' do
+      let(:user) { users(:affiliate_manager_with_not_approved_status) }
+
+      it 'raises an error' do
+        expect { finder.landing_page }.
+          to raise_error(LandingPageFinder::Error,
+                         LandingPageFinder::ACCESS_DENIED_TEXT)
```

Comment:
Let's be more explicit about the message we expect:
```ruby
      it 'raises an error' do
        expect { finder.landing_page }.
          to raise_error(LandingPageFinder::Error,
                         /Access Denied: These credentials are not recognized as valid/)
      end
```

Reply:
Changed.